pkg/orchestrator/assetscanestimationwatcher/watcher.go

```diff
@@ -0,0 +1,281 @@
+	if !ok {
+		if err = w.reconcileNoState(ctx, &assetScanEstimation); err != nil {
+			return err
+		}
+		return nil
+	}
+
+	logger.Tracef("Reconciling AssetScanEstimation state: %s", state)
+
+	switch state {
+	case models.AssetScanEstimationStateStatePending:
+		if err = w.reconcilePending(ctx, &assetScanEstimation); err != nil {
+			return err
+		}
+	case models.AssetScanEstimationStateStateAborted:
+		if err = w.reconcileAborted(ctx, &assetScanEstimation); err != nil {
+			return err
+		}
+	default:
+	}
+
+	return nil
+}
+
+func (w *Watcher) reconcileNoState(ctx context.Context, assetScanEstimation *models.AssetScanEstimation) error {
+	assetScanEstimationID, ok := assetScanEstimation.GetID()
+	if !ok {
+		return errors.New("invalid AssetScanEstimation: ID is nil")
+	}
+
+	assetScanEstimation.State.State = utils.PointerTo(models.AssetScanEstimationStateStatePending)
+	assetScanEstimation.State.LastTransitionTime = utils.PointerTo(time.Now())
+
+	assetScanEstimationPatch := models.AssetScanEstimation{
+		State: assetScanEstimation.State,
+	}
+	err := w.backend.PatchAssetScanEstimation(ctx, assetScanEstimationPatch, assetScanEstimationID)
+	if err != nil {
+		return fmt.Errorf("failed to update AssetScanEstimation. AssetScanEstimationID=%v: %w", assetScanEstimationID, err)
+	}
+	return nil
+}
+
+// nolint:cyclop
+func (w *Watcher) reconcilePending(ctx context.Context, assetScanEstimation *models.AssetScanEstimation) error {
+	logger := log.GetLoggerFromContextOrDiscard(ctx)
+
+	logger.Debugf("Reconciling pending asset scan estimations")
+
+	assetScanEstimationID, ok := assetScanEstimation.GetID()
+	if !ok {
+		return errors.New("invalid AssetScanEstimation: ID is nil")
+	}
+
+	if assetScanEstimation.Asset == nil || assetScanEstimation.Asset.AssetInfo == nil {
+		return errors.New("invalid AssetScanEstimation: Asset or AssetInfo is nil")
+	}
+
+	asset, err := assetScanEstimation.Asset.ToAsset()
+	if err != nil {
+		return fmt.Errorf("failed to convert assetRelationship to asset: %v", err)
+	}
+
+	stats := w.getLatestAssetScanStats(ctx, asset)
+
+	estimation, err := w.provider.Estimate(ctx, stats, asset, assetScanEstimation.AssetScanTemplate)
+
+	var fatalError provider.FatalError
+	var retryableError provider.RetryableError
+	switch {
+	case errors.As(err, &fatalError):
+		logger.Errorf("Failed to estimate asset scan: %v", err)
+		assetScanEstimation.State.State = utils.PointerTo(models.AssetScanEstimationStateStateFailed)
+		assetScanEstimation.State.StateMessage = utils.PointerTo(fatalError.Error())
+		assetScanEstimation.State.StateReason = utils.PointerTo(models.AssetScanEstimationStateStateReasonUnexpected)
+	case errors.As(err, &retryableError):
+		// nolint:wrapcheck
+		return common.NewRequeueAfterError(retryableError.RetryAfter(), retryableError.Error())
+	case err != nil:
+		logger.Errorf("Failed to estimate asset scan. Retrying after %v: %v", w.pollPeriod, err)
+		return common.NewRequeueAfterError(w.pollPeriod, err.Error())
```

Comment:
This should just be a normal error, and the Reconciler code will log it otherwise we'll be holding this in memory unnecessarily.